src/wasm/wasm-binary.cpp

```diff
@@ -376,14 +376,23 @@ void WasmBinaryWriter::writeGlobals() {
   }
   BYN_TRACE("== writeglobals\n");
   auto start = startSection(BinaryConsts::Section::Global);
-  auto num = importInfo->getNumDefinedGlobals();
+  Index num = 0;
+  ModuleUtils::iterDefinedGlobals(
+    *wasm, [&num](Global* global) { num += global->type.size(); });
```

Comment:
maybe a comment here too, that the number of globals in the binary is the sum of the globals after splitting up tuples?